doc/kuksa_requirementes.md

```diff
@@ -0,0 +1,805 @@
+  - [Data availability/persistence according to lifecycle of Client, Databroker and Provider](#data-availabilitypersistence-according-to-lifecycle-of-client-databroker-and-provider)
+  - [Wildcard support](#wildcard-support)
+  - [Registration of Datapoints](#registration-of-datapoints)
+  - [Availability of Datapoints](#availability-of-datapoints)
+  - [Lifecycle of components](#lifecycle-of-components)
+  - [Path requests](#path-requests)
+  - [Errors](#errors)
+  - [Setting Values](#setting-values)
+  - [Atomic operatoins](#atomic-operatoins)
+  - [Update notifications](#update-notifications)
+  - [Access rights](#access-rights)
+  - [VSS signals - Users vs providers](#vss-signals---users-vs-providers)
+  - [Enable "easy to use" user facing API](#enable-easy-to-use-user-facing-api)
+  - [Performance / runtime footprint](#performance--runtime-footprint)
+  - [Throttling-mode when system is overloaded.](#throttling-mode-when-system-is-overloaded)
+  - [Considerations regarding shared and zero-copy memory approaches](#considerations-regarding-shared-and-zero-copy-memory-approaches)
+  - [Provider control and provider capabilities](#provider-control-and-provider-capabilities)
+  - [Control the rate of updates](#control-the-rate-of-updates)
+  - [Differentiate between different providers of the same VSS data](#differentiate-between-different-providers-of-the-same-vss-data)
+  - [Data Aliveness/Availability](#data-alivenessavailability)
+  - [Missing features from `sdv.databroker.v1` in `kuksa.val.v1`](#missing-features-from-sdvdatabrokerv1-in-kuksavalv1)
+- [Exploring a design of a bidirectional streaming API](#exploring-a-design-of-a-bidirectional-streaming-api)
+    - [Design choices](#design-choices)
+        - [Stream](#stream)
+        - [Bidirectional stream](#bidirectional-stream)
+        - [Actuators](#actuators)
+    - [Overview](#overview)
+  - [COVESA topics](#covesa-topics)
+- [Domain requirements](#domain-requirements)
+  - [Interface requirements](#interface-requirements)
+    - [Current Databroker API design](#current-databroker-api-design)
+    - [New Databroker API proposal](#new-databroker-api-proposal)
+- [Appendix](#appendix)
+  - [1. Use Cases](#1-use-cases)
+  - [2. Protobuf API Diagram](#2-protobuf-api-diagram)
+
+# Documentation
+  ### [Terminology](./terminology.md)
+  ### [System Architecture](./system-architecture.md)
+
+# Functional requirements
+
+### As Signal Consumer
+1. As a client I want to have a single service point accessible from my execution environment.
+   
+2. As a client I want to `Get` sensor values or actuator current values of the vehicle and its environment in a uniform manner.
+  - Details:
+    * All sensor and actuator values for which a client has permission to.
+    * Client must provide the paths of the requested signals, which optionally may contain wildcards.
+    * I want to receive exactly all the signals’ values I requested without errors.
```

Comment:
Can we clarify this to something like: 
`I want to receive exactly all the signal values I requested or an error if retrieving at least on the requested signals fails` 
(not sure whether this is the current behavior)

Reply:
I don't think, that a consumer should rely on all the requested signals can be received without "errors". I wrote about states of signal (values) above. Some state might represent error situations, some just might be temporary lacking a valid value. I do **not** believe, that we can release consumers from handling those kind of situations (maybe there are some very simple consumers/app, where this would be possible. But more mature consumers will need - and want - to handle that failures.) 
Furthermore, if requesting multiple signals in one request. The failures shall be reported per signal separately.

Reply:
I think we aligned on that some time ago and understood that the decision was to return only a grpc error with detailed information on failed signals when dealing with a singles request to multiple signals as specified here (https://github.com/SoftwareDefinedVehicle/kuksa-databroker/blob/docs/kuksa_databroker_api/doc/kuksa_requirementes.md#errors).
So for me the requirement would be as you said @erikbosch:
`I want to receive exactly all the signal values I requested or an error if retrieving at least on the requested signals fails`